pkg/operator/ceph/webhook-config.go

```diff
@@ -0,0 +1,297 @@
+		},
+		Spec: api.IssuerSpec{
+			IssuerConfig: api.IssuerConfig{
+				SelfSigned: &api.SelfSignedIssuer{},
+			},
+		},
+	}
+	issuers, err := webhookContext.Issuers(namespace).Create(ctx, issuer, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create issuer %s. %v", issuerName, err)
+	}
+
+	sleepTime := 1
+	attempts := 10
+
+	for i := 0; i < attempts; i++ {
+		issuers, err = webhookContext.Issuers(namespace).Get(ctx, issuerName, metav1.GetOptions{})
+		if err != nil {
+			logger.Errorf("failed to get issuer %s. %v", issuerName, err)
+			return issuers, err
+		}
+
+		if len(issuers.Status.Conditions) != 0 {
+			if issuers.Status.Conditions[0].Status == v1.ConditionStatus(api.IssuerConditionReady) {
+				return issuers, nil
+			}
+		}
+		logger.Debugf("webhook config %q status=%+v", issuerName, issuers.Status.Conditions[0].Status)
+
+		time.Sleep(time.Duration(sleepTime) * time.Second)
+		logger.Infof("waiting for webhook config %q to be in ready status", issuerName)
+	}
+	return issuers, nil
+}
+
+func fetchorCreateCertificate(ctx context.Context, webhookContext *cs.CertmanagerV1Client, issuer *api.Issuer) error {
+	logger.Infof("Fetching certificate %s/%s.", namespace, certificateName)
+	_, err := webhookContext.Certificates(namespace).Get(ctx, certificateName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return createCertificate(ctx, webhookContext, issuer)
+		}
+		logger.Errorf("failed to get certificate %s. %v", certificateName, err)
+	}
+
+	return nil
+}
+
+func createCertificate(ctx context.Context, webhookContext *cs.CertmanagerV1Client, issuer *api.Issuer) error {
+	logger.Infof("Creating certificate %s/%s.", namespace, certificateName)
+	certificate := &api.Certificate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      certificateName,
+			Namespace: namespace,
+		},
+		Spec: api.CertificateSpec{
+			DNSNames: []string{
+				admissionControllerAppName,
+				fmt.Sprintf("%s.%s.svc", admissionControllerAppName, namespace),
+				fmt.Sprintf("%s.%s.svc.cluster.local", admissionControllerAppName, namespace)},
+			IssuerRef: v1.ObjectReference{
+				Name: issuer.Name,
+				Kind: "Issuer",
+			},
+			SecretName: admissionControllerAppName,
+		},
+	}
+
+	_, err := webhookContext.Certificates(namespace).Create(ctx, certificate, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create certificate %s. %v", certificateName, err)
+	}
+	sleepTime := 1
+	attempts := 10
+
+	for i := 0; i < attempts; i++ {
+		cert, err := webhookContext.Certificates(namespace).Get(ctx, certificateName, metav1.GetOptions{})
+		if err != nil {
+			logger.Errorf("failed to get certificate %s. %v", certificateName, err)
+		}
+		if len(cert.Status.Conditions) != 0 {
+			if cert.Status.Conditions[0].Status == v1.ConditionStatus(api.CertificateConditionReady) {
+				return nil
+			}
+		}
+		logger.Debugf("webhook config %q status=%+v", certificateName, cert.Status.Conditions[0].Status)
```

Comment:
If there are 0 conditions this will fail

Reply:
moved this block above in if condtion